### PR TITLE
allow transformEnvironment.sh to dereference env var values

### PR DESCRIPTION
**Pull request by tysonnorris**

### CODE REVIEW

**Comment by markusthoemmes** [comment]
Can you please provide a description of the change and why its needed? (I assume its needed for a different deployment topology?)

**Comment by tysonnorris** [comment]
e.g. we'd like to use a config value like `CONFIG_kamon_statsd_hostname $LIBPROCESS_IP`

**Comment by ddragosd** [comment]
LGTM. Once we move everything to PureConfig such behavior should come OOTB.

File: common/scala/transformEnvironment.sh [hunk 1]
```diff
@@ -21,6 +21,18 @@ props=()
 for var in $configVariables
 do
     value=$(printenv "$var")
+    #allow us to derefence environment variables, e.g. CONFIG_some_key=$SOME_ENV_VAR
```
Comment (markusthoemmes) [inline common/scala/transformEnvironment.sh]: Typo `dereference`

File: common/scala/transformEnvironment.sh [hunk 2]
```diff
@@ -21,6 +21,18 @@ props=()
...
+    if [[ $value == \$* ]]
```
Comment (markusthoemmes) [inline common/scala/transformEnvironment.sh]: A comment would be good: `# iff the value starts with $`

File: common/scala/transformEnvironment.sh [hunk 3]
```diff
@@ -21,6 +21,18 @@ props=()
...
+    then
+        varname=${value:1}
```
Comment (markusthoemmes) [inline common/scala/transformEnvironment.sh]: comment: `# drop the starting '$'`

File: common/scala/transformEnvironment.sh [hunk 4]
```diff
@@ -21,6 +21,18 @@ props=()
...
+        value2=${!varname}
```
Comment (markusthoemmes) [inline common/scala/transformEnvironment.sh]: - comment `# '!' dereferences the variable`
- rename `value2` to `dereferencedValue`?

**Comment by tysonnorris** [comment]
thanks @markusthoemmes - comments added :)

**Pull request merged**